examples/chat/module/rust/src/lib.rs

```diff
@@ -14,138 +14,125 @@
 // limitations under the License.
 //
 
-use log::info;
+use log::{error, info, warn};
 use oak::{
     grpc,
-    io::{Sender, SenderExt},
+    io::{ReceiverExt, Sender, SenderExt},
+    Label,
 };
 use oak_abi::proto::oak::application::ConfigMap;
-use proto::{
-    command::Command::{JoinRoom, SendMessage},
-    Chat, ChatDispatcher, Command, CreateRoomRequest, DestroyRoomRequest, SendMessageRequest,
-    SubscribeRequest,
-};
-use std::collections::{hash_map::Entry, HashMap};
+use proto::{Chat, ChatDispatcher, Message, SendMessageRequest, SubscribeRequest};
+use std::collections::HashMap;
 
-mod backend;
 mod proto {
     include!(concat!(env!("OUT_DIR"), "/oak.examples.chat.rs"));
 }
 
-type RoomId = Vec<u8>;
-type AdminToken = Vec<u8>;
-
+/// A node that routes each incoming gRPC invocation to a per-room worker node (either pre-existing,
+/// or newly created) that can handle requests with the label of the incoming request.
+///
+/// This node never looks at the contents of the invocation messages, only at the labels of its
+/// channels, and therefore keeps a public confidentiality label, which also allows it to create
+/// further nodes and channels, with more specific labels.
 #[derive(Default)]
-struct Node {
-    rooms: HashMap<RoomId, Room>,
+struct Router {
+    /// Maps each label to a channel to a dedicated worker node for that label, corresponding to
+    /// the `room` entry point of this module.
+    rooms: HashMap<Label, Sender<oak::grpc::Invocation>>,
+}
+
+impl oak::CommandHandler<oak::grpc::Invocation> for Router {
+    fn handle_command(&mut self, command: oak::grpc::Invocation) -> Result<(), oak::OakError> {
+        // The router node has a public confidentiality label, and therefore cannot read the
+        // contents of the request of the invocation (unless it happens to be public as well), but
+        // it can always inspect its label.
+        match &command.receiver {
+            Some(receiver) => {
+                let label = receiver.label()?;
+                // Check if there is a channel to a room with the desired label already, or create
+                // it if not.
+                let channel = self.rooms.entry(label.clone()).or_insert_with(|| {
+                    let (wh, rh) = oak::io::channel_create_with_label(&label)
+                        .expect("could not create channel");
+                    oak::node_create_with_label(
+                        &oak::node_config::wasm("app", "room"),
+                        &label,
```

Comment:
Does this work? I was not able to create a node with any label other than the public-untrusted, because of this line:
https://github.com/project-oak/oak/blob/dd849816917f72ea585a191c975e4caab4438e1b/oak_runtime/src/lib.rs#L757